index.bs

```diff
@@ -2704,15 +2745,191 @@ partial interface MLGraphBuilder {
+            1. Let |size| be the [=operand-shape=][|index|] of |input|.
+            1. If |size| is not equal to 1 and not equal to |newShape|[index], then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+            1. If |size| is equal to 1, then let |outputDesc|.{{MLOperandDescriptor/dimensions}}[|index|] be |newShape|[|index|].
+    1. If any of the following sub-steps fail, [=exception/throw=] an "{{OperationError}}" {{DOMException}}.
+        1. Let |output| be the result of <a>creating an MLOperand</a> given [=this=] and |outputDesc|.
+        1. Make a request to the underlying platform to:
+            1. Create an [=implementation-defined=] platform operator |expandImpl| for this method, given |input| and |newShape|.
+            1. Store a reference of |expandImpl| in |output|.{{MLOperand/[[operator]]}}.
+            1. Create an [=implementation-defined=] platform operand |outputImpl| to represent output,given |output| and |expandImpl|.
+            1. Store a reference to |outputImpl| in |output|.{{MLOperand/[[operand]]}}.
+        1. Connect |input| as input to |expandImpl|.
+        1. Connect |output|.{{MLOperand/[[operand]]}} as output to |expandImpl|.
+    1. Return |output|.
+  </div>
+</details>
+
+### gather ### {#api-mlgraphbuilder-gather}
+Gather values of the input tensor along an axis according to the indices.
+<script type=idl>
+dictionary MLGatherOptions {
+  unsigned long axis = 0;
+};
+
+partial interface MLGraphBuilder {
+  MLOperand gather(MLOperand input, MLOperand indices, optional MLGatherOptions options = {});
+};
+</script>
+
+{{MLGatherOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLGatherOptions>
+    : <dfn>axis</dfn>
+    ::
+        An {{unsigned long}} scalar specifying the axis along which the gathered values are obtained. Its value must be in the range [0, N-1] where N is the [=rank=] of the input tensor.
+</dl>
+
+<div>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input N-D tensor from which the values are gathered.
+        - *indices*: an {{MLOperand}}. The indices N-D tensor of the input values to gather. The values must be of type "uint32" in the range [0, N-1] where N is the size of the input dimension indexed by *options.axis*.
+        - *options*: an optional {{MLGatherOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output N-D tensor of [=rank=] equal to M + N - 1 where M and N are the [=rank=] of the *input* and *indices* tensor respectively.
+</div>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>gather(|input|, |indices|, |options|)</dfn> method steps are:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |input| and |indices| is {{MLOperand}}.
+    1. Let |shapeInput| be |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} and |rankInput| the [=list/size=] of |shapeInput|.
```

Comment:
Please modify
`and |rankInput| the [=list/size=] of |shapeInput|.`
to
`and |rankInput| be the [=list/size=] of |shapeInput|.`

Reply:
The line still reads grammatically correctly to me without adding the *be*.

Reply:
Even though it's wordier, using consistent phrasing (e.g. "if ... then ...", "let ... be ...", "set ... to ...", etc) helps searching and find/replace on the spec.

Reply:
I agree search-ability is important and especially so in the algorithmic parts of the spec.

These algorithms are read by human compilers, and a few such human compilers already emitted warning suggesting this phrasing needs to be fixed :)

@wchao1115 once fixed, please mark this conversation as resolved.

Reply:
Fixed.